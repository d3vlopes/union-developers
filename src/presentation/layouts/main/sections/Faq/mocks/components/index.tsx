import { vitest } from 'vitest'

vitest.mock('@/presentation/components/organisms', () => ({
  QuestionsAccordion: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return (
      <div data-testid="mock-questions-accordion">{children}</div>
    )
  },
}))
