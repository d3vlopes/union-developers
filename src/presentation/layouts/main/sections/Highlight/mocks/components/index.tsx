import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  Highlights: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return <div data-testid="mock-highlights">{children}</div>
  },
}))
