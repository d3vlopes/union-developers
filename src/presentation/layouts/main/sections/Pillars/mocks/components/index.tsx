import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  MethodologyCards: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return <div data-testid="mock-methodology-cards">{children}</div>
  },
}))
