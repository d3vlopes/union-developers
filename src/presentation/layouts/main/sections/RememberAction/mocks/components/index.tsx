import { vitest } from 'vitest'

vitest.mock(
  '@/presentation/components/molecules/RememberAction',
  () => ({
    RememberAction: function Mock({
      children,
    }: {
      children: React.ReactNode
    }) {
      return <div data-testid="mock-remember-action">{children}</div>
    },
  }),
)
