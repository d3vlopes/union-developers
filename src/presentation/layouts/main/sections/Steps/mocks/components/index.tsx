import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  Steps: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="mock-steps">{children}</div>
  },
}))
