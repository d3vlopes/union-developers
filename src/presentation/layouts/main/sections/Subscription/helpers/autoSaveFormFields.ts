import { Dispatch, SetStateAction } from 'react'

export function autoSaveFormFields<TFormValues, TFormData>(
  formValues: TFormValues,
  formData: TFormData,
  errors: object,
  setFormData: Dispatch<SetStateAction<TFormData>>,
  delay = 5000,
) {
  const interval = setInterval(() => {
    const isFieldContainError = Object.keys(errors).length > 0

    if (!isFieldContainError) {
      setFormData({ ...formData, ...formValues })
    }
  }, delay)

  return interval
}
