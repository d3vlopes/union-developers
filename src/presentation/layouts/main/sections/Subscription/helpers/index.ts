export * from './autoSaveFormFields'
export * from './formStepMap'
