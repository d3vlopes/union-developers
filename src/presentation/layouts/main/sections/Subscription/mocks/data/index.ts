import { generateId } from '@/utils'

import { termsMock } from '../../steps/AboutYou/components/Terms/mock'

import { SubscriptionSectionProps } from '../..'

export const subscriptionMock: SubscriptionSectionProps = {
  id: 'formulario',
  title: 'Junte-se ao time!',
  initialStepDescription:
    'Tem interesse em participar do Union? Confira abaixo os requisitos para participar',
  formStepsDescription:
    'Ficamos felizes em saber que você tem interesse em fazer parte do Union! Agora é só responder o nosso formulário',
  requirements: [
    'Ter mais que 18 anos',
    'Ter 1 ano ou mais de estudo em programação',
    'Ainda não trabalhar na área',
    'Já ter criado projetos sozinho que não sejam de cursos ou eventos',
    'Não participar de bootcamps ou programas de capacitação',
    'Ter disponibilidade para trabalhar no mínimo 1 hora por dia no projeto',
    'Ter disponibilidade para participar das reuniões do time, durante a semana e finais de semana',
  ],
  buttonText: 'Inscrever-se',
  formStepNavLinks: [
    {
      id: generateId(),
      stepTitle: 'Dados pessoais',
      stepNumber: 1,
    },
    {
      id: generateId(),
      stepTitle: 'Informações técnicas',
      stepNumber: 2,
    },
    {
      id: generateId(),
      stepTitle: 'Sobre você',
      stepNumber: 3,
    },
    {
      id: generateId(),
      stepTitle: 'Enviar',
      stepNumber: 4,
    },
  ],
  terms: termsMock,
}
