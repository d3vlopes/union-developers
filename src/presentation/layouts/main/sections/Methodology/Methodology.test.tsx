import { render, screen } from '@/__tests__/helpers'

import { mainLayoutMock } from '@/presentation/layouts/main/mocks/data'

import './mocks'

import { MethodologySection } from '.'

const { getByRole, getByText, getByTestId } = screen

describe('<MethodologySection />', () => {
  it('should render heading', () => {
    render(
      <MethodologySection {...mainLayoutMock.methodologySection} />,
    )

    const heading = getByRole('heading', { name: 'Metodologias' })

    expect(heading).toBeInTheDocument()
  })

  it('should render description', () => {
    render(
      <MethodologySection {...mainLayoutMock.methodologySection} />,
    )

    const description = getByText(
      mainLayoutMock.methodologySection.description,
    )

    expect(description).toBeInTheDocument()
  })

  it('should render methodology cards', () => {
    render(
      <MethodologySection {...mainLayoutMock.methodologySection} />,
    )

    const methodologyCards = getByTestId('mock-methodology-cards')

    expect(methodologyCards).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(
      <MethodologySection {...mainLayoutMock.methodologySection} />,
    )

    expect(container).toMatchSnapshot()
  })
})
