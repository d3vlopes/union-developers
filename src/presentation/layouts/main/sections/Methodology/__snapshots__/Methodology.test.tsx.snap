// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MethodologySection /> > should match snapshot 1`] = `
.c0 {
  position: relative;
  margin-top: 7.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #1F1D2B;
  background-clip: padding-box;
  border: 3px solid transparent;
  border-radius: 20px;
}

.c0:before {
  content: '';
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: -1;
  margin: -3px;
  border-radius: inherit;
  background: linear-gradient(-74deg,#0048FE 45%,#851D86 100%);
}

.c0:after {
  content: '';
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: -1;
  border-radius: inherit;
  margin: -5px;
  background: linear-gradient( 360deg, #1F1D2B 95%, rgba(31,29,43,0) 100% );
}

.c1 {
  position: absolute;
  display: inline-block;
  background: #1F1D2B;
  margin: -18px 0 0 calc(100vw - 85%);
  padding: 0 15px 0 15px;
}

.c1 h2 {
  font-size: 2rem;
  font-weight: 600;
  background: linear-gradient(225deg,#0048FE -50%,#851D86 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  text-align: center;
}

.c2 {
  padding: 2.4rem 1.6rem;
  margin: 0 auto;
}

.c2 p {
  color: #E2E2E2;
  text-align: center;
  line-height: 150%;
}

.c2 p:not(:last-child) {
  padding-bottom: 2.4rem;
}

.c3 {
  margin-top: 4rem;
}

@media (min-width:768px) {
  .c0 {
    margin-top: calc(3.2rem + 11rem);
  }

  .c0:before {
    background: linear-gradient(-74deg,#0048FE -10%,#851D86 100%);
  }

  .c0:after {
    background: linear-gradient( 360deg, #1F1D2B 75%, rgba(31,29,43,0) 100% );
  }
}

@media (min-width:320px) {
  .c1 {
    margin: -18px 0 0 calc(100vw - 91%);
  }
}

@media (min-width:365px) {
  .c1 {
    margin: -18px 0 0 calc(100vw - 85%);
  }
}

@media (min-width:428px) {
  .c1 {
    margin: -18px 0 0 calc(100vw - 82%);
  }
}

@media (min-width:768px) {
  .c1 {
    margin: -35px 0 0 30%;
  }

  .c1 h2 {
    font-size: 4rem;
  }
}

@media (min-width:1024px) {
  .c1 {
    margin: -35px 0 0 37%;
  }

  .c1 h2 {
    font-size: 4rem;
  }
}

@media (min-width:768px) {
  .c2 {
    padding: 5rem 8rem 5rem;
  }

  .c2 p {
    max-width: 102.7rem;
    font-size: 2.4rem;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-top: 8rem;
  }
}

<div>
  <section
    class="c0"
    id="methodologys"
  >
    <header
      class="c1"
    >
      <h2>
        Metodologias
      </h2>
    </header>
    <div
      class="c2"
    >
      Na Union combinamos o uso de 3 metodologias para você conquistar experiência, saber se posicionar profissionalmente e acelerar sua evolução!
    </div>
    <div
      class="c3"
    >
      <div
        data-testid="mock-methodology-cards"
      />
    </div>
  </section>
</div>
`;
